billcom_integration/models/res_partner.py

```diff
@@ -0,0 +1,677 @@
+# Copyright 2025 Binhex - Simple Solutions
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _name = "res.partner"
+    _inherit = ["res.partner", "billcom.abstract.model"]
+
+    billcom_res_currency_id = fields.Many2one(
+        "res.currency",
+        string="Bill.com Currency",
+        default=lambda self: self.env.company.currency_id,
+    )
+
+    billcom_sync_state = fields.Selection(
+        [
+            ("pending", "Pending Sync"),
+            ("synced", "Synced"),
+            ("error", "Sync Error"),
+        ],
+        default="pending",
+        string="Bill.com Sync State",
+    )
+
+    billcom_payment_purpose_id = fields.Many2one(
+        "billcom.payment.purpose",
+        string="Bill.com Payment Purpose",
+        help="Payment purpose for international vendors (non-US) in Bill.com",
+        domain="[('country_id', '=', country_id)]",
+    )
+
+    def _prepare_partner_data(self, partner_type="vendor"):  # noqa: C901
+        """Prepare partner data for Bill.com API v3"""
+        self.ensure_one()
+        if not self.is_sync_to_billcom:
+            return False
+
+        # Get billing and shipping addresses
+        billing_address = self
+        shipping_address = self
+
+        # If this is a contact with a parent company, use parent for some fields
+        # parent_company = (
+        #     self.parent_id if self.parent_id and self.parent_id.is_company else False
+        # )
+
+        # Check for specific address types in child contacts
+        if self.child_ids:
+            for child in self.child_ids:
+                if child.type == "invoice":
+                    billing_address = child
+                elif child.type == "delivery":
+                    shipping_address = child
```

Comment:
```suggestion
        for child in self.child_ids:
            if child.type == "invoice":
                billing_address = child
            elif child.type == "delivery":
                shipping_address = child
```
A partner could have multiple invoice and delivery address so ...